src/cli/lint/internalLints/refreshConfigFiles.ts

```diff
@@ -12,37 +12,47 @@ import { createDestinationFileReader } from '../../configure/analysis/project';
 import { mergeWithIgnoreFile } from '../../configure/processing/ignoreFile';
 import type { InternalLintResult } from '../internal';
 
-const REFRESHABLE_IGNORE_FILES = [
-  '.eslintignore',
-  '.gitignore',
-  '.prettierignore',
+const ensureNoNpmrcExclusion = (s: string) =>
+  s
+    .split('\n')
+    .filter((line) => line.includes('!') || !line.includes('.npmrc'))
+    .join('\n');
+
+const ensureNoAuthToken = (s: string) =>
+  s
+    .split('\n')
+    .filter((line) => !line.includes('authToken'))
+    .join('\n');
+
+const REFRESHABLE_CONFIG_FILES = [
+  { name: '.eslintignore' },
+  { name: '.gitignore', additionalMapping: ensureNoNpmrcExclusion },
+  { name: '.prettierignore' },
+  { name: '.npmrc', additionalMapping: ensureNoAuthToken },
```

Comment:
Would it make sense to make this refresh/management conditional on the package manager? https://github.com/seek-oss/skuba/blob/ed2e36ad22498640977da1216fe46388837f338d/src/utils/packageManager.ts#L37-L39

I think that would rule out any potential backward incompatibilities with Yarn projects, which are still the overwhelming majority in SEEK-Jobs.

Reply:
Pushed an attempt - probably needs scrutiny